massgen/subagent/launch_watcher.py

```
@@ -0,0 +1,367 @@
+            )
+            status = "completed" if exit_code == 0 else "error"
+        except TimeoutError:
+            logger.warning(f"[SubagentLaunchWatcher] Subagent {subagent_id} timed out")
+            status = "timeout"
+            exit_code = -1
+        except asyncio.CancelledError:
+            logger.warning(f"[SubagentLaunchWatcher] Subagent {subagent_id} cancelled")
+            status = "cancelled"
+            exit_code = -1
+            raise  # Re-raise so the task is marked cancelled
+        except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)